cardano-cli/src/Cardano/CLI/Run/Legacy/Query.hs

```diff
@@ -255,18 +255,20 @@ runQueryProtocolParameters socketPath (AnyConsensusModeParams cModeParams) netwo
     eInMode <- toEraInMode era cMode
       & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
 
-    lift (queryProtocolParameters eInMode sbe)
+    result <- lift (queryProtocolParameters eInMode sbe)
       & onLeft (left . ShelleyQueryCmdUnsupportedNtcVersion)
       & onLeft (left . ShelleyQueryCmdEraMismatch)
 
+    pure (InAnyEra era result)
+
   writeProtocolParameters mOutFile =<< except (join (first ShelleyQueryCmdAcquireFailure result))
 
  where
   writeProtocolParameters
     :: Maybe (File () Out)
-    -> ProtocolParameters
+    -> InAnyEra era ProtocolParameters
     -> ExceptT ShelleyQueryCmdError IO ()
-  writeProtocolParameters mOutFile' pparams =
+  writeProtocolParameters mOutFile' (InAnyEra _ pparams) =
```

Comment:
Here, even though we are outside the scope of the computation that produced the `pparams`, we are able to have access to the same `era` type that was available within that computation's context.